packages/api-i18n/src/graphql/graphql/locales.ts

```diff
@@ -124,12 +128,16 @@ const plugin: GraphQLSchemaPlugin<I18NContext & SecurityContext> = {
                     if (data.default) {
                         await i18n.locales.updateDefault(data.code);
                     }
-
-                    return new Response(data);
+                    const locale = await i18n.locales.getByCode(data.code);
```

Comment:
If we'd migrate the logic into crud file, we'd probably be able to avoid this extra DB call.